### ISSUE TITLE
Plotting custom types and structs as values

### ISSUE THREAD

**Issue by CrustyAuklet** [awulkiew/graphical-debugging#53]
I have a type that looks something like this:
```
```c++
class CAttenEntry {
    float   m_atten = -1.0f;
public:
    CAttenEntry() : m_atten( -1 ) {}
    float    atten() const         { return m_atten >= 0 ? m_atten : 0.5f; }
    bool    calculated() const  { return m_atten >= 0; }
    operator double() const   {  return atten(); }
};
```
I added the `operator double()` function to try to get the value to plot in this extension, but I can't get it to work so far. The values are in a C-style array, but I have tried copying the values into a vector and that still doesn't work. As of now I am copying the `atten()` values to a vector every time the array changes and that "works", but is very slow.

Using C++ and Visual Studio 2019

**Comment by awulkiew** [comment]
Right. The vscode extension supports user-defined values but this one doesn't. This should be added here as well.

I don't know why it doesn't work with conversion operator. It's probably a bug. Even it it was working, the visualization would be slow because it'd be based on strings returned by the debugger instead of accessing the memory directly. So memory-access should still be added.

With that said. the development for VS2019 is discontinued so when implemented this will be available in VS2022.

**Comment by awulkiew** [comment]
It seems that the conversion operator doesn't work in debugger. I can't remember if it worked in older versions of VS or that I assumed it will. E.g. for this struct:
```
struct MyValue
{
    operator double() const { return v; }
    double v = 0;
};
```

I get the following in Watch:
  | Name | Value | Type
-- | -- | -- | --
  | double(my_value) | no suitable conversion function from "MyValue" to "double" exists |  |

Is it the same for you?

It seems that no only user defined values should be supported explicitly but also documentation altered.
